### ISSUE TITLE
Recognising shifted holidays

### ISSUE THREAD

**Issue by ShaheedHaque** [workalendar/workalendar#691]
It would be useful to know if a holiday whose observance is shifted can be easily recognised as such. For example, if "Boxing Day" is observed on "Boxing Day Shift", is the use of the suffix characters " Shift" something that can be **reliably** used to derive its original name?

I appreciate that string-hacking is not much of an API, but if needed, a better option might be to use a subclass of "str" something like this:
```
>>> class Shifted(str):
...     SUFFIX = ' Shift'
...     SUFFIX_LENGTH = len(SUFFIX)
...     
...     def __new__(cls, value):
...         obj = str.__new__(cls, value + cls.SUFFIX)
...         return obj
...     
...     def origin(self):
...         return self[:-self.SUFFIX_LENGTH]
... 
>>> shifted = Shifted('Boxing Day')
>>> shifted
'Boxing Day Shift'
>>> isinstance(shifted, str)
True
>>> isinstance(shifted, Shifted)
True
>>> shifted.origin()
'Boxing Day'
```
I would find such a thing very useful. Of course, isinstance() isn't the world's greatest API, but one could go further if desired using str -> Holiday -> Shifted in place of the current str-based holiday name?

**Comment by ewjoachim** [comment]
I think we might need a different API where holidays would be objects, not strings. (dataclasses would probably be nice). We would then be able to query different attributes of the objects such as the date, whether it's shifted, maybe event the translated name etc.

**Comment by ShaheedHaque** [comment]
> we might need a different API where holidays would be objects

My suggestion was purely driven by the anticipated need to keep backwards compatibility. With the approach I outlined: no existing user would see any change, while allowing both the kind of simple additions I am interested and, possibly, easing the path to introducing i18n/l10n.

Of course, if that is not a constraint, then a more conventional approach could be a better choice.

**Comment by ewjoachim** [comment]
Yeah but it's highly dependent on the consistency of human-readable names, and depending on how we implement localization, it may completely break.

I'm not advocating for a change in the existing API but on a new API. Say one simple API for simple cases, and one advanced API, slightly more complex to use but allowing for more things. Also with greater chances to be future proof.
